Task springmvc

Sindicat007 opened the pull request.

Sindicat007 committed:
Для проверки

Signed-off-by: Sindicat007 <sindicat007@ya.ru>

Sindicat007 committed:
Изменил версию Спринга в помнике

Signed-off-by: Sindicat007 <sindicat007@ya.ru>

RinatBeybutov commented on SpringMvc/src/main/java/ru/maxima/springmvc/repository/BookRepository.java:
```diff
@@ -0,0 +1,37 @@
+
+    private final SessionFactory sessionFactory;
+
+    public BookRepository(SessionFactory sessionFactory) {
```
ломбок здесь не подойдет?

RinatBeybutov commented on SpringMvc/src/main/java/ru/maxima/springmvc/repository/BookRepository.java:
```diff
@@ -0,0 +1,37 @@
+    public BookRepository(SessionFactory sessionFactory) {
+        this.sessionFactory = sessionFactory;
+    }
+@Transactional
```
отступы поправь и пробел между методами

RinatBeybutov commented on SpringMvc/src/main/java/ru/maxima/springmvc/service/BookService.java:
```diff
@@ -0,0 +1,27 @@
+public class BookService {
+    private final BookRepository bookRepository;
+
+    public BookService(BookRepository bookRepository) {
```
ломбок

RinatBeybutov commented on SpringMvc/src/main/java/ru/maxima/springmvc/service/BookService.java:
```diff
@@ -0,0 +1,27 @@
+        this.bookRepository = bookRepository;
+    }
+
+    @Transactional
```
не нужно вешать транзакцию на сервис и репозиторий. Оставь где-то в одном месте, но так, чтобы все связанные операции были под одной транзакцией

Sindicat007 committed:
- Поправил отступы
- Добавил конструкторы через lombok
- Убрал дублирование транзакция в BookService

Signed-off-by: Sindicat007 <sindicat007@ya.ru>

RinatBeybutov commented on SpringMvc/src/main/java/ru/maxima/springmvc/controller/MvcController.java:
```diff
@@ -0,0 +1,48 @@
+
+    private final BookService bookService;
+
+    public MvcController(BookService bookService) {
```
убрать этот конструктор и добавить аннотацию @RequiredArgsConstructor

RinatBeybutov commented on SpringMvc/src/main/java/ru/maxima/springmvc/repository/BookRepository.java:
```diff
@@ -0,0 +1,38 @@
+
+import java.util.List;
+
+@AllArgsConstructor
```
@requiredargsconstructor

RinatBeybutov commented on SpringMvc/src/main/java/ru/maxima/springmvc/controller/MvcController.java:
```diff
@@ -0,0 +1,48 @@
+    }
+
+    @GetMapping("/showBooks")
+    public String showBooks(Model model) {
```
давай методы, связанные с книгами давай вынесем в отдельный контроллер

RinatBeybutov commented on SpringMvc/src/main/java/ru/maxima/springmvc/service/BookService.java:
```diff
@@ -0,0 +1,24 @@
+        bookRepository.saveBook(book);
+    }
+
+    @Transactional
```
дублирование транзакции здесь и в репозитории

Sindicat007 committed:
- Создал отдельный контроллер для book
- Заменил аннотации для создания конструкторов
- Убрал дублирование транзакций

Signed-off-by: Sindicat007 <sindicat007@ya.ru>
